### PR TITLE
Fix build windows

### PR DESCRIPTION
**Pull request by mattn**
* M_PI is not defined in C standard. Refer to: https://stackoverflow.com/questions/26065359/m-pi-flagged-as-undeclared-identifier
* Convert string to wstring since SharedLibrary::LoadLibrary take wstring.

### CODE REVIEW

File: tensorflow/core/lib/random/random_distributions_utils.h [hunk 1]
```diff
@@ -22,6 +22,10 @@ limitations under the License.
 
 #include "tensorflow/core/lib/random/philox_random.h"
 
+#ifndef M_PI
```
Comment (MeghnaNatraj) [inline tensorflow/core/lib/random/random_distributions_utils.h:25]: Why is this change needed?

Comment (mattn) [reply tensorflow/core/lib/random/random_distributions_utils.h:25]: M_PI is not defined in C standard. M_PI is currently used only in this file.

**Review by MeghnaNatraj — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

File: tensorflow/lite/delegates/external/external_delegate.cc [hunk 2]
```diff
@@ -29,10 +30,12 @@ struct ExternalLib {
       const char**, const char**, size_t,
       void (*report_error)(const char*))>::type;
   using DestroyDelegatePtr = std::add_pointer<void(TfLiteDelegate*)>::type;
+  struct wchar_codecvt : public std::codecvt<wchar_t, char, std::mbstate_t> {};
+  std::wstring_convert<wchar_codecvt> converter;
 
   // Open a given delegate library and load the create/destroy symbols
   bool load(const std::string library) {
-    void* handle = SharedLibrary::LoadLibrary(library.c_str());
+    void* handle = SharedLibrary::LoadLibrary(converter.from_bytes(library.c_str()).c_str());
```
Comment (MeghnaNatraj) [inline tensorflow/lite/delegates/external/external_delegate.cc:38]: All the newly added code in this file needs to be within the _WIN32 header guard:

eg:
#if defined(_WIN32)
  void* handle = SharedLibrary::LoadLibrary(converter.from_bytes(library.c_str()).c_str());
#else
 void* handle = SharedLibrary::LoadLibrary(library.c_str());
#endif  // defined(_WIN32)

Refer to lite/shared_library.h for details.

Without this update, the PR cannot be merged as the code will fail on non-windows machines (refer to the failing CI build errors for this PR)

**Review by MeghnaNatraj — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

File: tensorflow/core/lib/random/random_distributions_utils.h (hunk 1 above)
Comment (cloudhan) [reply tensorflow/core/lib/random/random_distributions_utils.h:25]: @MeghnaNatraj This is not the correct way to introduce math constant, you are mis-clarified!

@mattn  See MSVC document of [Math Constants](https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170) to see how to do it in sane way.
